src/goto-instrument/cover_filter.cpp

```diff
@@ -43,33 +44,33 @@ bool internal_functions_filtert::operator()(
   return true;
 }
 
-/// Filter functions whose name match the regex
-/// \param function_id: a function name
+/// Filter functions whose name matches the regex
+/// \param function: the function under consideration
 /// \param goto_function: a goto function
 /// \return returns true if the function name matches
 bool include_pattern_filtert::operator()(
-  const irep_idt &function_id,
+  const symbolt &function,
   const goto_functionst::goto_functiont &goto_function) const
 {
   (void)goto_function; // unused parameter
   std::smatch string_matcher;
-  return std::regex_match(
-    id2string(function_id), string_matcher, regex_matcher);
+  return std::regex_match(id2string(function.name), string_matcher, regex_matcher);
 }
 
 /// Call a goto_program non-trivial if it has:
 ///  * Any declarations
 ///  * At least 2 branches
 ///  * At least 5 assignments
 /// These criteria are arbitrarily chosen.
-/// \param function_id: name of \p goto_function
+/// \param function: function symbol for function corresponding
+/// to \p goto_function
 /// \param goto_function: a goto function
 /// \return returns true if non-trivial
 bool trivial_functions_filtert::operator()(
-  const irep_idt &function_id,
+  const symbolt &function,
   const goto_functionst::goto_functiont &goto_function) const
 {
-  (void)function_id; // unused parameter
+  (void)function; // unused parameter
```

Comment:
:pick: If unused just delete the name of the field.

Reply:
Note that you can't refer to it from doxygen then...

Reply:
Why would you want to document an unused parameter?

Reply:
@thk123 the way we’ve set up linting we require all parameters to a function to be documented if any are